ath79: Add support for CPE210 v1, CPE510v1, CPE220 v2

adschm opened the pull request:
This adds support for CPE210 v1, CPE510v1, CPE220v2 after some preparations (tplink-safeloader partition adjustments and DTSI reorganization).

**This is tested and ready for merge.**

~~The device boots and has working WiFi, _but for ethernet I need help with the correct setup._
Note that this is a two-port device compared to most of the other CPExxx devices which are one-port (CPE210 v2+, CPE510 v2+, CPE610v1).~~
~~There has already been a discussion for the very similar CPE510 v1 in https://github.com/openwrt/openwrt/pull/1704, but work on it stopped in-between so I fail to draw helpful conclusions on the correct setup.~~
Ethernet is working now (thanks to 981213)

Side-note:
I have checked tplink-safeloader with automatic partition splitting (tplink-safeloader and tplink,firmware) as well as uimage (tplink-safeloader-uimage and denx,uimage) without success as already reported on the other CPE devices with ar9344.

adrianschmutzler commented:
@apcameron Can you check the shared DTSI on your CPE510/610? Should be sufficient if you only pick patch no. 2. (However, note that I have not even build-tested yet).

@p-wassi @apcameron @981213 
May you aid me on what I have to do to set up ethernet on this device?

981213 commented:
@adrianschmutzler You could try to copy ethernet setup from Qihoo C301.

981213 commented:
A correct result should be that one port is on builtin switch and the other one is eth1.
If copying dts from C301 makes two ethernet ports become port 1 and port 4 on builtin switch, you'd need the follwing instead:
```
&eth1 {
	status = "okay";
	gmac-config {
		device = <&gmac>;
		switch-phy-swap = <0>;
		switch-only-mode = <1>;
	};
};

&eth0 {
	status = "okay";
	phy-handle = <&swphy4>;
};

```

blocktrron commented on target/linux/ath79/dts/ar9344_tplink_cpe210-v1.dts:
```diff
@@ -0,0 +1,77 @@
+};
+
+&gpio {
+	status = "okay";
```
`status` property should be safe to drop. The GPIO controller has it's status not set to disabled in the inherited dtsi.

adrianschmutzler replied:
Yes, will do it with the next update.

blocktrron reviewed (commented):
Regarding the mtdsplit: This is currently not working as the kernel is prepended by the LZMA loader.

We might however be able to either add a `kernel-offset` property for this case or even detect this circumstance automatically. I will look into this later.

adrianschmutzler commented:
@blocktrron Note that mtdsplit IS working on CPE210 v2/v3 (qca9533).

blocktrron commented:
@adrianschmutzler Both the CPE210v2 and CPE210v3 don't use the LZMA loader prepended to the kernel image, so the mtdsplit driver can read proper offsets from the first header.

apcameron commented:
@adrianschmutzler  @robimarko is also looking at the TP-LINK CPE210 V1 and I believe has it working on the ath79 branch so you may want to work with him on this. As far as your changes for the TP-LINK CPE610 v1 and CPE510 V2/V3 I can try to test a build within the next week or so time permitting.
Note:-  the TP-LINK CPE210 V1.1 AND TP-LINK CPE220 V2 and  almost identical to the  TP-LINK CPE210 V1 and so the code to include them should be trivial. I have the TP-LINK CPE220 V2 working on the ar71xx branch see the trivial change I made at https://github.com/aredn/aredn_ar71xx/blob/develop/patches/001-add_support_for_TP-Link_CPE220_v2.patch

adrianschmutzler commented:
Thanks @981213 , ethernet port are working now (`switch-phy-swap = <0>` is required).

However, I have some remaining issues:
- lan1 LED (eth0) is enabled and blinking after boot
- Sometimes, connecting LAN-Port on PoE-Adapter (device: eth1) leads to reboot of the device

adrianschmutzler commented:
@apcameron So the CPE220 V2 is identical to the CPE210 V1?

apcameron commented:
@adrianschmutzler  it appears to be almost identical. I do not have one but I helped someone in New Zealand get it working. Note The CPE210 V1.1 and CPE220 V2 sometimes have a problem being recovered and what I discovered is it looks for vmlinuz when recovering and this is what I did
I compiled a RAM Version of the software which we renamed to vmlinuz and placed it in the tftp folder  and it booted right up. See
https://community.tp-link.com/en/business/forum/topic/150131?page=1

We were then able to use scp to upload a full version of the software and use sysupgrade -F -v -n /tmp/filename.bin to reflash it.
 
The T-LINK CPE220 V3 which I am working on is similar to the CPE210 V2 but with 2 ethernet ports.

981213 commented:
> lan1 LED (eth0) is enabled and blinking after boot (if nothing connected)

This is expected. eth0 connects to port 0 of builtin switch so it's always link up. You can use swconfig trigger on switch port 4 instead.

> Sometimes, (dis)connecting LAN-Port on PoE-Adapter (device: eth1) leads to reboot of the device

No idea about this one :(

adrianschmutzler commented:
@apcameron 
> it appears to be almost identical. I do not have one but I helped someone in New Zealand get it working.

If I add it here (which looks trivial), can you have the 220v2 tested?
220 is the 210 with removable antennas?

> The CPE210 V1.1 and CPE220 V2 sometimes have a problem being recovered

I know :-( They seem to ship different bootloaders (or whatever). I bricked the first lended CPE210 I received and then had to remove the chip and reflash it to get it working again. TFTP recovery never worked on this one. (I did not try vmlinuz)
I then acquired another CPE210 and with this one TFTP just works without any issues. Both are V1.1.

adschm committed:
tplink-safeloader: increase kernel partition for CPE/WBSx10v1

This is a preparation for ath79 support of the CPE210/CPE510 v1.
Kernel size is chosen equal to the latest update for CPE610 v1.

This also updates the partition size in ar71xx target, so code
remains consistent if someone looks up the device. Since CPE210,
CPE510, WBS210 and WBS510 (all v1) share the same partition
layout definition, and are on deprecated target anyway, this
changes them all at once.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
ath79: provide common DTSI for CPE510 v2/v3 and CPE610

This puts some common code into a new shared DTSI. Common nodes
are chosen so that the new DTSI can be used for CPE210 v1, too.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adrianschmutzler commented:
> This is expected. eth0 connects to port 0 of builtin switch so it's always link up. You can use swconfig trigger on switch port 4 instead.

Works fine now (just pushed the final state), network config seems to be set up correctly now. Is there any kind of objective guideline which port should be WAN and which LAN by default?

I still have the reboot-problem. (Switched power adapter and cable to be sure.)

adrianschmutzler commented:
I have added support for CPE510 v1 on top. The corresponding PR#1704 has not seen updates since January, and adding the device here is trivial.
However, I do not own this device, so someone else would have to test it (otherwise the commit can be removed on merge).
@p-wassi Maybe you want to test this one...

Considering CPE210v1 and CPE510v1, DTSes have considerable amount of shared code. I'm undecided on whether I should create another cpexxx-v1.dtsi for those (will be a lot of nesting then ...)

apcameron commented:
@adrianschmutzler If I remember from comments by @ynezz in an earlier PR #1748 for the CPE610-v1 all references to
**lan0** or **LAN0** should be replaced by lan and LAN

As far as testing the CPE220 V2 I will include you in a e-mail to the person who tested it for me in the past.

adrianschmutzler commented:
Hmm, if I do not (un)plug cables, it does not reboot.
dmesg looks normal (eth0/eth1 are also swapped here):
```
root@OpenWrt:~# dmesg
[    0.000000] Linux version 4.19.56 (adsc@buildfff) (gcc version 7.4.0 (OpenWrt GCC 7.4.0 r10451-653e05d27f)) #0 Mon Jul 8 15:01:54 2019
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 0001974c (MIPS 74Kc)
[    0.000000] MIPS: machine is TP-LINK CPE210 v1
[    0.000000] SoC: Atheros AR9344 rev 2
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 04000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] On node 0 totalpages: 16384
[    0.000000]   Normal zone: 144 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 16384 pages, LIFO batch:3
[    0.000000] random: get_random_bytes called from start_kernel+0x98/0x4a4 with crng_init=0
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 16240
[    0.000000] Kernel command line: console=ttyS0,115200 rootfstype=squashfs,jffs2
[    0.000000] Dentry cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Inode-cache hash table entries: 4096 (order: 2, 16384 bytes)
[    0.000000] Writing ErrCtl register=00000000
[    0.000000] Readback ErrCtl register=00000000
[    0.000000] Memory: 57512K/65536K available (4305K kernel code, 174K rwdata, 1016K rodata, 1216K init, 207K bss, 8024K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 51
[    0.000000] CPU clock: 560.000 MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 6825930166 ns
[    0.000011] sched_clock: 32 bits at 280MHz, resolution 3ns, wraps every 7669584382ns
[    0.009078] Calibrating delay loop... 278.93 BogoMIPS (lpj=1394688)
[    0.086307] pid_max: default: 32768 minimum: 301
[    0.091925] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.099625] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.113612] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.125093] futex hash table entries: 256 (order: -1, 3072 bytes)
[    0.132348] pinctrl core: initialized pinctrl subsystem
[    0.139641] NET: Registered protocol family 16
[    0.152154] GPIO line 18 (tp-link:ext:lna0) hogged as output/high
[    0.159303] GPIO line 19 (tp-link:ext:lna1) hogged as output/high
[    0.196317] clocksource: Switched to clocksource MIPS
[    0.203684] NET: Registered protocol family 2
[    0.209825] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 4096 bytes)
[    0.218858] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    0.226991] TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
[    0.234388] TCP: Hash tables configured (established 1024 bind 1024)
[    0.241947] UDP hash table entries: 256 (order: 0, 4096 bytes)
[    0.248811] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
[    0.256558] NET: Registered protocol family 1
[    0.261653] PCI: CLS 0 bytes, default 32
[    0.265887] Crashlog allocated RAM at address 0x3f00000
[    0.273812] workingset: timestamp_bits=14 max_order=14 bucket_order=0
[    0.289544] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.296377] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.324617] io scheduler noop registered
[    0.329250] io scheduler deadline registered (default)
[    0.337735] pinctrl-single 1804002c.pinmux: 544 pins, size 68
[    0.345574] Serial: 8250/16550 driver, 1 ports, IRQ sharing disabled
[    0.353891] console [ttyS0] disabled
[    0.358153] 18020000.uart: ttyS0 at MMIO 0x18020000 (irq = 9, base_baud = 2500000) is a 16550A
[    0.368197] console [ttyS0] enabled
[    0.375611] bootconsole [early0] disabled
[    0.403804] m25p80 spi0.0: s25fl064k (8192 Kbytes)
[    0.408772] 6 fixed-partitions partitions found on MTD device spi0.0
[    0.415220] Creating 6 MTD partitions on "spi0.0":
[    0.420119] 0x000000000000-0x000000020000 : "u-boot"
[    0.426192] 0x000000020000-0x000000030000 : "partition-table"
[    0.433135] 0x000000030000-0x000000040000 : "info"
[    0.439051] 0x000000040000-0x0000007c0000 : "firmware"
[    0.445203] 2 fixed-partitions partitions found on MTD device firmware
[    0.451899] Creating 2 MTD partitions on "firmware":
[    0.456964] 0x000000000000-0x000000200000 : "kernel"
[    0.462980] 0x000000200000-0x000000780000 : "rootfs"
[    0.469010] mtd: device 5 (rootfs) set to be root filesystem
[    0.480081] 1 squashfs-split partitions found on MTD device rootfs
[    0.486437] 0x000000480000-0x000000780000 : "rootfs_data"
[    0.492922] 0x0000007c0000-0x0000007f0000 : "config"
[    0.498998] 0x0000007f0000-0x000000800000 : "art"
[    0.506080] libphy: Fixed MDIO Bus: probed
[    0.867419] ag71xx 19000000.eth: Could not connect to PHY device. Deferring probe.
[    1.546753] libphy: ag71xx_mdio: probed
[    1.551852] libphy: ar8xxx-mdio: probed
[    1.563080] switch0: Atheros AR8229 rev. 1 switch registered on mdio-bus.0
[    1.615389] ag71xx 1a000000.eth: connected to PHY at fixed-0:00 [uid=00000000, driver=Generic PHY]
[    1.625323] eth0: Atheros AG71xx at 0xba000000, irq 5, mode: gmii
[    1.633543] GACT probability on
[    1.636816] u32 classifier
[    1.639562]     input device check on
[    1.643274]     Actions configured
[    1.648062] NET: Registered protocol family 10
[    1.659953] Segment Routing with IPv6
[    1.663819] NET: Registered protocol family 17
[    1.668497] 8021q: 802.1Q VLAN Support v1.8
[    2.008311] ag71xx 19000000.eth: connected to PHY at mdio-bus.0:1f:04 [uid=004dd042, driver=Generic PHY]
[    2.019071] eth1: Atheros AG71xx at 0xb9000000, irq 4, mode: mii
[    2.035774] VFS: Mounted root (squashfs filesystem) readonly on device 31:5.
[    2.051533] Freeing unused kernel memory: 1216K
[    2.056129] This architecture does not have kernel memory protection.
[    2.062698] Run /sbin/init as init process
[    2.546331] random: fast init done
```

apcameron commented:
@adrianschmutzler if you check you e-mail you will see you now have a Tester for the TP-LINK CPE220 V2.
You may have to build the image for him and send it via google drive as that I how I did it in the past

adrianschmutzler commented:
Looks like something is triggering the reboot:
```
Sat Jul 20 01:00:55 2019 daemon.info procd: - shutdown -
Sat Jul 20 01:00:56 2019 authpriv.info dropbear[903]: Early exit: Terminated by signal
Sat Jul 20 01:00:56 2019 authpriv.info dropbear[2459]: Exit (root): Terminated by signal
Sat Jul 20 01:01:00 2019 authpriv.info dropbear[905]: Not backgrounding
```

apcameron commented:
@adrianschmutzler  If the Ports seem to be the wrong way around try adding
the following to 02_network

        tplink,cpe210-v1)
                ucidef_set_interfaces_lan_wan "eth1" "eth0"
                ;;
for all the devices with 2 ports in this PR

apcameron commented:
@adrianschmutzler I have created a build based on your PR and done a quick test.

TP-LINK CPE610 v1 works as before
TP-LINK CPE510 v2 works as before
TP-LINK CPE510 v3 I do not have one to test currently
TP-LINK CPE210 v1.1 see below
I had to add the "tplink,cpe210-v1|\"  entry to target/linux/ath79/base-files/etc/board.d/02_network to get the ports the right way around.

        comfast,cf-e110n-v2|\
        comfast,cf-e120a-v3|\
        tplink,cpe210-v1|\
        ubnt,nanostation-m|\
        ubnt,routerstation)
                ucidef_set_interfaces_lan_wan "eth1" "eth0"
                ;;

adrianschmutzler commented:
@apcameron Thanks for testing.

> ucidef_set_interfaces_lan_wan "eth1" "eth0"

I'm aware of that choice.

However, how do you define the "right way" here? Is this somehow implied by vendor firmware?

adschm committed:
ath79: create common definition tplink-loader-okli

The loader-okli is shared by several TP-Link CPExxx devices, so
give it its own definition to prevent too much code duplication.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
ath79: add support for TP-Link CPE210 v1

Specifications:

    * SoC: Qualcomm Atheros AR9344 (560 MHz)
    * RAM: 64MB
    * Storage: 8 MB
    * Wireless: 2.4GHz N based built into SoC 2x2
    * Ethernet: 2x 100/10 Mbps, integrated into SoC, 24V POE IN

Installation:

Flash factory image through stock firmware WEB UI
or through TFTP:
To get to TFTP recovery just hold reset button while powering on for
around 4-5 seconds and release.
Rename factory image to recovery.bin
Stock TFTP server IP:192.168.0.100
Stock device TFTP address:192.168.0.254

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
ath79: add support for TP-Link CPE510 v1

TP-Link CPE510-v1 is an outdoor wireless CPE for 5 GHz with
two Ethernet ports based on Atheros AR9334

Specifications:
 - 560/450/225 MHz (CPU/DDR/AHB)
 - 2x 10/100 Mbps Ethernet, 1x PoE-in, 1x PoE-out
 - 64 MB of DDR2 RAM
 - 8 MB of SPI-NOR Flash
 - 2T2R 5 GHz
 - 13 dBi built-in antenna
 - Power, LAN0, LAN1 green LEDs
 - 4x green RSSI LEDs

Flash factory image through stock firmware WEB UI
or through TFTP:
To get to TFTP recovery just hold reset button while powering on for
around 4-5 seconds and release.
Rename factory image to recovery.bin
Stock TFTP server IP:192.168.0.100
Stock device TFTP address:192.168.0.254

Based on the work of Paul Wassi <p.wassi@gmx.at>

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
ath79: add support for TP-Link CPE220 v2

This seems to be identical to CPE210 v1 despite having removable
antennas.

Specifications:

    * SoC: Qualcomm Atheros AR9344 (560 MHz)
    * RAM: 64MB
    * Storage: 8 MB
    * Wireless: 2.4GHz N based built into SoC 2x2
    * Ethernet: 2x 100/10 Mbps, integrated into SoC, 24V POE IN

Installation:

Flash factory image through stock firmware WEB UI
or through TFTP:
To get to TFTP recovery just hold reset button while powering on for
around 4-5 seconds and release.
Rename factory image to recovery.bin
Stock TFTP server IP:192.168.0.100
Stock device TFTP address:192.168.0.254

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

apcameron commented:
@adrianschmutzler It is so that for existing users port LAN0 still is the port that provides an ip to the device that connects to it.

adrianschmutzler commented:
Update:

1. Added a common DTSI for the two-port devices, so the DTSes now only contain compatible and model.
2. Added a definition "tplink_loader_okli" in common-tp-link.mk, so all ar9344 CPE devices can inherit from it.
3. Swapped LAN/WAN (@apcameron I personally prefer it the way it was before, but if you say it's some kind of OpenWrt standard to keep client on LAN0 I will follow that)
4. Remove some CPE510V2 string from tplink-safeloader definition of CPE510V1
5. Added CPE220v2 (@apcameron Please have a look at code (="review") here first. I copied the device specs from CPE210v1 for the commit message, can you verify them?)

Note that commits are a little scattered in the conversation tab, so maybe use commit tab to prevent overlooking something.

>  If I remember from comments by @ynezz in an earlier PR #1748 for the CPE610-v1 all references to lan0 or LAN0 should be replaced by lan and LAN

I do not interpret it this way, but only with effect on what you did with CPE610. I would keep it as it is right now, reviewer can still complain about it. In worst case, it is easy to change and would be sufficient to be tested on a single device. (We do not merge LED definitions for one-port and two-port, so it does not matter anyway IMO.

Concerning my reboot issues:
I can only reproduce them if I repeatedly connect and disconnect a cable at the LAN port of the PoE adapter. If the device is just running with or without cables connected, it is perfectly stable.

apcameron commented:
@adrianschmutzler With regards to the reboots what happens if you use another PoE Adapter with the same rating?

With regards to Model numbers in the past on ar71xx they were setup in target/linux/ar71xx/base-files/lib/ar71xx.sh and extracted from the devices in some cases.
The Convention was model="TP-Link xxxxxx vx.x"
EG **TP-Link CPE510 v1.0**


Going forward I can see that we have shortened the version
but we have a discrepancy in the TP-Link vs TP-LINK as its not consistent.
See below
New
model = "**TP-LINK** CPE210 v1";
model = "**TP-LINK** CPE220 v2";
model = "**TP-LINK** CPE510 v1";
model = "**TP-Link** CPE610 v1";
model = "**TP-Link** CPE510 v2";
model = "**TP-Link** CPE510 v3";

Now would be the time do decide on the final naming and fix the ones that differ.

I suggest reviewing the DTS files for other TP-LINK Devices as they seem to use **TP-Link**
EG ar7240_tplink_tl-wr743nd-v1.dts ar9331_tplink_tl-mr3040-v2.dts  qca9533_tplink_tl-wr841-v9.dts all use **TP-Link** Not **TP-LINK**

I will re-build my images in the next day or so to test the latest updates and ask Eddie
to test the code I build from this for the CPE220 V2

adrianschmutzler commented:
@apcameron 

> With regards to the reboots what happens if you use another PoE Adapter with the same rating?

Checked with different PoE adapter and LAN cable already. Might be courageous enough to test my other CPE210 with broken TFTP soon.

> a discrepancy in the TP-Link vs TP-LINK as its not consistent

I will change to "TP-Link" (just an old copy/paste error from my side). This has been used in the recent introduction of DEVICE_VENDOR in https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/image/common-tp-link.mk#L51

While grepping through ath79, only the CPE210 v2 and v3 are also using all-caps. I will push a change to those here, too.

However, I will wait with both capitalization changes until tests are done, as they won't affect functionality.

adrianschmutzler commented:
BTW: Thanks for testing and good find regarding the name capitalization.

adschm committed:
ath79: fix vendor capitalization of TP-Link

According to detective grep, with this patch all devices should
be labelled "TP-Link" consistently.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

apcameron commented:
@adrianschmutzler I will try to do more testing tomorrow or the day after. Hopefully we will get feedback from @ae6xe in the next day or so too on his testing on the TP-Link CPE510 v1.1

ae6xe commented:
Not finding any issues on cpe510 v1.1

I made a minor mod in /etc/config/network to use 192.168.0.1 to not collide with my home network.  laptop on LAN0 and home network on LAN1 and entering this comment. 

[    0.000000] Linux version 4.19.57 (joe@AE6XE-VM) (gcc version 7.4.0 (OpenWrt GCC 7.4.0 r10560+5-04cb651376)) #0 Sun Jul 21 03:13:21 2019
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 0001974c (MIPS 74Kc)
[    0.000000] MIPS: machine is TP-LINK CPE510 v1
[    0.000000] SoC: Atheros AR9344 rev 2
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 04000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] On node 0 totalpages: 16384
[    0.000000]   Normal zone: 144 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 16384 pages, LIFO batch:3
[    0.000000] random: get_random_bytes called from start_kernel+0x98/0x4a8 with crng_init=0
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 16240
[    0.000000] Kernel command line: console=ttyS0,115200 rootfstype=squashfs,jffs2
[    0.000000] Dentry cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Inode-cache hash table entries: 4096 (order: 2, 16384 bytes)
[    0.000000] Writing ErrCtl register=00000000
[    0.000000] Readback ErrCtl register=00000000
[    0.000000] Memory: 57576K/65536K available (4266K kernel code, 177K rwdata, 1008K rodata, 1196K init, 207K bss, 7960K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 51
[    0.000000] CPU clock: 560.000 MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 6825930166 ns
[    0.000010] sched_clock: 32 bits at 280MHz, resolution 3ns, wraps every 7669584382ns
[    0.009076] Calibrating delay loop... 278.93 BogoMIPS (lpj=1394688)
[    0.086308] pid_max: default: 32768 minimum: 301
[    0.091933] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.099633] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.113658] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.125140] futex hash table entries: 256 (order: -1, 3072 bytes)
[    0.132393] pinctrl core: initialized pinctrl subsystem
[    0.139672] NET: Registered protocol family 16
[    0.152011] GPIO line 18 (tp-link:ext:lna0) hogged as output/high
[    0.159167] GPIO line 19 (tp-link:ext:lna1) hogged as output/high
[    0.195741] clocksource: Switched to clocksource MIPS
[    0.203070] NET: Registered protocol family 2
[    0.209158] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 4096 bytes)
[    0.218181] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    0.226308] TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
[    0.233695] TCP: Hash tables configured (established 1024 bind 1024)
[    0.241261] UDP hash table entries: 256 (order: 0, 4096 bytes)
[    0.248120] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
[    0.255906] NET: Registered protocol family 1
[    0.261007] PCI: CLS 0 bytes, default 32
[    0.265346] Crashlog allocated RAM at address 0x3f00000
[    0.273270] workingset: timestamp_bits=14 max_order=14 bucket_order=0
[    0.288877] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.295650] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.324324] io scheduler noop registered
[    0.328952] io scheduler deadline registered (default)
[    0.337463] pinctrl-single 1804002c.pinmux: 544 pins, size 68
[    0.345294] Serial: 8250/16550 driver, 1 ports, IRQ sharing disabled
[    0.353594] console [ttyS0] disabled
[    0.357847] 18020000.uart: ttyS0 at MMIO 0x18020000 (irq = 9, base_baud = 2500000) is a 16550A
[    0.367891] console [ttyS0] enabled
[    0.375308] bootconsole [early0] disabled
[    0.394986] m25p80 spi0.0: gd25q64 (8192 Kbytes)
[    0.399808] 6 fixed-partitions partitions found on MTD device spi0.0
[    0.406289] Creating 6 MTD partitions on "spi0.0":
[    0.411166] 0x000000000000-0x000000020000 : "u-boot"
[    0.417250] 0x000000020000-0x000000030000 : "partition-table"
[    0.424076] 0x000000030000-0x000000040000 : "info"
[    0.429974] 0x000000040000-0x0000007c0000 : "firmware"
[    0.436206] 2 fixed-partitions partitions found on MTD device firmware
[    0.442841] Creating 2 MTD partitions on "firmware":
[    0.447939] 0x000000000000-0x000000200000 : "kernel"
[    0.453938] 0x000000200000-0x000000780000 : "rootfs"
[    0.459944] mtd: device 5 (rootfs) set to be root filesystem
[    0.470870] 1 squashfs-split partitions found on MTD device rootfs
[    0.477230] 0x000000400000-0x000000780000 : "rootfs_data"
[    0.483715] 0x0000007c0000-0x0000007f0000 : "config"
[    0.489787] 0x0000007f0000-0x000000800000 : "art"
[    0.496934] libphy: Fixed MDIO Bus: probed
[    0.856844] ag71xx 19000000.eth: Could not connect to PHY device. Deferring probe.
[    1.536199] libphy: ag71xx_mdio: probed
[    1.541278] libphy: ar8xxx-mdio: probed
[    1.552513] switch0: Atheros AR8229 rev. 1 switch registered on mdio-bus.0
[    1.604815] ag71xx 1a000000.eth: connected to PHY at fixed-0:00 [uid=00000000, driver=Generic PHY]
[    1.614748] eth0: Atheros AG71xx at 0xba000000, irq 5, mode: gmii
[    1.624137] NET: Registered protocol family 10
[    1.635832] Segment Routing with IPv6
[    1.639701] NET: Registered protocol family 17
[    1.644300] 8021q: 802.1Q VLAN Support v1.8
[    1.987733] ag71xx 19000000.eth: connected to PHY at mdio-bus.0:1f:04 [uid=004dd042, driver=Generic PHY]
[    1.998491] eth1: Atheros AG71xx at 0xb9000000, irq 4, mode: mii
[    2.013416] VFS: Mounted root (squashfs filesystem) readonly on device 31:5.
[    2.029048] Freeing unused kernel memory: 1196K
[    2.033651] This architecture does not have kernel memory protection.
[    2.040218] Run /sbin/init as init process
[    2.555757] random: fast init done
[    3.075995] init: Console is alive
[    3.079740] init: - watchdog -
[    4.410802] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    4.592537] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    4.610466] init: - preinit -
[    5.852939] random: jshn: uninitialized urandom read (4 bytes read)
[    6.038760] random: jshn: uninitialized urandom read (4 bytes read)
[    6.178010] random: jshn: uninitialized urandom read (4 bytes read)
[    6.782420] IPv6: ADDRCONF(NETDEV_UP): eth1: link is not ready
[    8.908127] eth1: link up (100Mbps/Full duplex)
[    8.912769] IPv6: ADDRCONF(NETDEV_CHANGE): eth1: link becomes ready
[   10.130982] jffs2: notice: (478) jffs2_build_xattr_subsystem: complete building xattr subsystem, 3 of xdatum (0 unchecked, 1 orphan) and 7 of xref (1 dead, 0 orphan) found.
[   10.148925] mount_root: switching to jffs2 overlay
[   10.180203] overlayfs: upper fs does not support tmpfile.
[   10.194593] urandom-seed: Seeding with /etc/urandom.seed
[   10.402877] eth1: link down
[   10.420395] procd: - early -
[   10.423446] procd: - watchdog -
[   11.074172] procd: - watchdog -
[   11.077787] procd: - ubus -
[   11.111908] urandom_read: 5 callbacks suppressed
[   11.111918] random: ubusd: uninitialized urandom read (4 bytes read)
[   11.131076] random: ubusd: uninitialized urandom read (4 bytes read)
[   11.139264] procd: - init -
[   11.972552] kmodloader: loading kernel modules from /etc/modules.d/*
[   12.021525] Loading modules backported from Linux version v5.2-rc7-0-g6fbc7275c7a9
[   12.029285] Backport generated by backports.git v5.2-rc7-1-0-g021a6ba1
[   12.101722] xt_time: kernel timezone is -0000
[   12.139668] urngd: v1.0.0 started.
[   12.279028] PPP generic driver version 2.4.2
[   12.296999] NET: Registered protocol family 24
[   12.430306] ath: EEPROM regdomain: 0x0
[   12.430316] ath: EEPROM indicates default country code should be used
[   12.430321] ath: doing EEPROM country->regdmn map search
[   12.430338] ath: country maps to regdmn code: 0x3a
[   12.430345] ath: Country alpha2 being used: US
[   12.430349] ath: Regpair used: 0x3a
[   12.446223] ieee80211 phy0: Selected rate control algorithm 'minstrel_ht'
[   12.448210] ieee80211 phy0: Atheros AR9340 Rev:2 mem=0xb8100000, irq=12
[   12.563703] random: crng init done
[   12.567848] kmodloader: done loading kernel modules from /etc/modules.d/*
[   22.999848] br-lan: port 1(eth1) entered blocking state
[   23.005165] br-lan: port 1(eth1) entered disabled state
[   23.010908] device eth1 entered promiscuous mode
[   23.035618] IPv6: ADDRCONF(NETDEV_UP): br-lan: link is not ready
[   23.100138] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
[   24.186965] eth0: link up (1000Mbps/Full duplex)
[   24.191697] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[   25.145623] eth1: link up (100Mbps/Full duplex)
[   25.323378] br-lan: port 1(eth1) entered blocking state
[   25.328749] br-lan: port 1(eth1) entered forwarding state
[   25.355889] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
root@OpenWrt:~# dmesg
[    0.000000] Linux version 4.19.57 (joe@AE6XE-VM) (gcc version 7.4.0 (OpenWrt GCC 7.4.0 r10560+5-04cb651376)) #0 Sun Jul 21 03:13:21 2019
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 0001974c (MIPS 74Kc)
[    0.000000] MIPS: machine is TP-LINK CPE510 v1
[    0.000000] SoC: Atheros AR9344 rev 2
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 04000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] On node 0 totalpages: 16384
[    0.000000]   Normal zone: 144 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 16384 pages, LIFO batch:3
[    0.000000] random: get_random_bytes called from start_kernel+0x98/0x4a8 with crng_init=0
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 16240
[    0.000000] Kernel command line: console=ttyS0,115200 rootfstype=squashfs,jffs2
[    0.000000] Dentry cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Inode-cache hash table entries: 4096 (order: 2, 16384 bytes)
[    0.000000] Writing ErrCtl register=00000000
[    0.000000] Readback ErrCtl register=00000000
[    0.000000] Memory: 57576K/65536K available (4266K kernel code, 177K rwdata, 1008K rodata, 1196K init, 207K bss, 7960K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 51
[    0.000000] CPU clock: 560.000 MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 6825930166 ns
[    0.000010] sched_clock: 32 bits at 280MHz, resolution 3ns, wraps every 7669584382ns
[    0.009076] Calibrating delay loop... 278.93 BogoMIPS (lpj=1394688)
[    0.086308] pid_max: default: 32768 minimum: 301
[    0.091933] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.099633] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.113658] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.125140] futex hash table entries: 256 (order: -1, 3072 bytes)
[    0.132393] pinctrl core: initialized pinctrl subsystem
[    0.139672] NET: Registered protocol family 16
[    0.152011] GPIO line 18 (tp-link:ext:lna0) hogged as output/high
[    0.159167] GPIO line 19 (tp-link:ext:lna1) hogged as output/high
[    0.195741] clocksource: Switched to clocksource MIPS
[    0.203070] NET: Registered protocol family 2
[    0.209158] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 4096 bytes)
[    0.218181] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    0.226308] TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
[    0.233695] TCP: Hash tables configured (established 1024 bind 1024)
[    0.241261] UDP hash table entries: 256 (order: 0, 4096 bytes)
[    0.248120] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
[    0.255906] NET: Registered protocol family 1
[    0.261007] PCI: CLS 0 bytes, default 32
[    0.265346] Crashlog allocated RAM at address 0x3f00000
[    0.273270] workingset: timestamp_bits=14 max_order=14 bucket_order=0
[    0.288877] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.295650] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.324324] io scheduler noop registered
[    0.328952] io scheduler deadline registered (default)
[    0.337463] pinctrl-single 1804002c.pinmux: 544 pins, size 68
[    0.345294] Serial: 8250/16550 driver, 1 ports, IRQ sharing disabled
[    0.353594] console [ttyS0] disabled
[    0.357847] 18020000.uart: ttyS0 at MMIO 0x18020000 (irq = 9, base_baud = 2500000) is a 16550A
[    0.367891] console [ttyS0] enabled
[    0.375308] bootconsole [early0] disabled
[    0.394986] m25p80 spi0.0: gd25q64 (8192 Kbytes)
[    0.399808] 6 fixed-partitions partitions found on MTD device spi0.0
[    0.406289] Creating 6 MTD partitions on "spi0.0":
[    0.411166] 0x000000000000-0x000000020000 : "u-boot"
[    0.417250] 0x000000020000-0x000000030000 : "partition-table"
[    0.424076] 0x000000030000-0x000000040000 : "info"
[    0.429974] 0x000000040000-0x0000007c0000 : "firmware"
[    0.436206] 2 fixed-partitions partitions found on MTD device firmware
[    0.442841] Creating 2 MTD partitions on "firmware":
[    0.447939] 0x000000000000-0x000000200000 : "kernel"
[    0.453938] 0x000000200000-0x000000780000 : "rootfs"
[    0.459944] mtd: device 5 (rootfs) set to be root filesystem
[    0.470870] 1 squashfs-split partitions found on MTD device rootfs
[    0.477230] 0x000000400000-0x000000780000 : "rootfs_data"
[    0.483715] 0x0000007c0000-0x0000007f0000 : "config"
[    0.489787] 0x0000007f0000-0x000000800000 : "art"
[    0.496934] libphy: Fixed MDIO Bus: probed
[    0.856844] ag71xx 19000000.eth: Could not connect to PHY device. Deferring probe.
[    1.536199] libphy: ag71xx_mdio: probed
[    1.541278] libphy: ar8xxx-mdio: probed
[    1.552513] switch0: Atheros AR8229 rev. 1 switch registered on mdio-bus.0
[    1.604815] ag71xx 1a000000.eth: connected to PHY at fixed-0:00 [uid=00000000, driver=Generic PHY]
[    1.614748] eth0: Atheros AG71xx at 0xba000000, irq 5, mode: gmii
[    1.624137] NET: Registered protocol family 10
[    1.635832] Segment Routing with IPv6
[    1.639701] NET: Registered protocol family 17
[    1.644300] 8021q: 802.1Q VLAN Support v1.8
[    1.987733] ag71xx 19000000.eth: connected to PHY at mdio-bus.0:1f:04 [uid=004dd042, driver=Generic PHY]
[    1.998491] eth1: Atheros AG71xx at 0xb9000000, irq 4, mode: mii
[    2.013416] VFS: Mounted root (squashfs filesystem) readonly on device 31:5.
[    2.029048] Freeing unused kernel memory: 1196K
[    2.033651] This architecture does not have kernel memory protection.
[    2.040218] Run /sbin/init as init process
[    2.555757] random: fast init done
[    3.075995] init: Console is alive
[    3.079740] init: - watchdog -
[    4.410802] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    4.592537] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    4.610466] init: - preinit -
[    5.852939] random: jshn: uninitialized urandom read (4 bytes read)
[    6.038760] random: jshn: uninitialized urandom read (4 bytes read)
[    6.178010] random: jshn: uninitialized urandom read (4 bytes read)
[    6.782420] IPv6: ADDRCONF(NETDEV_UP): eth1: link is not ready
[    8.908127] eth1: link up (100Mbps/Full duplex)
[    8.912769] IPv6: ADDRCONF(NETDEV_CHANGE): eth1: link becomes ready
[   10.130982] jffs2: notice: (478) jffs2_build_xattr_subsystem: complete building xattr subsystem, 3 of xdatum (0 unchecked, 1 orphan) and 7 of xref (1 dead, 0 orphan) found.
[   10.148925] mount_root: switching to jffs2 overlay
[   10.180203] overlayfs: upper fs does not support tmpfile.
[   10.194593] urandom-seed: Seeding with /etc/urandom.seed
[   10.402877] eth1: link down
[   10.420395] procd: - early -
[   10.423446] procd: - watchdog -
[   11.074172] procd: - watchdog -
[   11.077787] procd: - ubus -
[   11.111908] urandom_read: 5 callbacks suppressed
[   11.111918] random: ubusd: uninitialized urandom read (4 bytes read)
[   11.131076] random: ubusd: uninitialized urandom read (4 bytes read)
[   11.139264] procd: - init -
[   11.972552] kmodloader: loading kernel modules from /etc/modules.d/*
[   12.021525] Loading modules backported from Linux version v5.2-rc7-0-g6fbc7275c7a9
[   12.029285] Backport generated by backports.git v5.2-rc7-1-0-g021a6ba1
[   12.101722] xt_time: kernel timezone is -0000
[   12.139668] urngd: v1.0.0 started.
[   12.279028] PPP generic driver version 2.4.2
[   12.296999] NET: Registered protocol family 24
[   12.430306] ath: EEPROM regdomain: 0x0
[   12.430316] ath: EEPROM indicates default country code should be used
[   12.430321] ath: doing EEPROM country->regdmn map search
[   12.430338] ath: country maps to regdmn code: 0x3a
[   12.430345] ath: Country alpha2 being used: US
[   12.430349] ath: Regpair used: 0x3a
[   12.446223] ieee80211 phy0: Selected rate control algorithm 'minstrel_ht'
[   12.448210] ieee80211 phy0: Atheros AR9340 Rev:2 mem=0xb8100000, irq=12
[   12.563703] random: crng init done
[   12.567848] kmodloader: done loading kernel modules from /etc/modules.d/*
[   22.999848] br-lan: port 1(eth1) entered blocking state
[   23.005165] br-lan: port 1(eth1) entered disabled state
[   23.010908] device eth1 entered promiscuous mode
[   23.035618] IPv6: ADDRCONF(NETDEV_UP): br-lan: link is not ready
[   23.100138] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
[   24.186965] eth0: link up (1000Mbps/Full duplex)
[   24.191697] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[   25.145623] eth1: link up (100Mbps/Full duplex)
[   25.323378] br-lan: port 1(eth1) entered blocking state
[   25.328749] br-lan: port 1(eth1) entered forwarding state
[   25.355889] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready

root@OpenWrt:~# swconfig dev switch0 show
Global attributes:
	enable_vlan: 0
	ar8xxx_mib_poll_interval: 500
	ar8xxx_mib_type: 0
	enable_mirror_rx: 0
	enable_mirror_tx: 0
	mirror_monitor_port: 0
	mirror_source_port: 0
	arl_table: address resolution table
Port 0: MAC d4:6e:0e:32:c1:c2
Port 4: MAC 68:72:51:0b:8a:07
Port 4: MAC 28:16:2e:c4:7b:a9
Port 4: MAC 00:00:54:fe:07:cc

Port 0:
	mib: MIB counters
RxGoodByte  : 1368122 (1.3 MiB)
TxByte      : 16076619 (15.3 MiB)

	pvid: 0
	link: port:0 link:up speed:1000baseT full-duplex txflow rxflow 
Port 1:
	mib: No MIB data
	pvid: 0
	link: port:1 link:down
Port 2:
	mib: No MIB data
	pvid: 0
	link: port:2 link:down
Port 3:
	mib: No MIB data
	pvid: 0
	link: port:3 link:down
Port 4:
	mib: MIB counters
RxGoodByte  : 16076619 (15.3 MiB)
TxByte      : 1368122 (1.3 MiB)

	pvid: 0
	link: port:4 link:up speed:100baseT full-duplex auto

br-lan    Link encap:Ethernet  HWaddr D4:6E:0E:32:C1:C2  
          inet addr:192.168.0.1  Bcast:192.168.0.255  Mask:255.255.255.0
          inet6 addr: fd48:2a2a:6ce7::1/60 Scope:Global
          inet6 addr: fe80::d66e:eff:fe32:c1c2/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:9943 errors:0 dropped:0 overruns:0 frame:0
          TX packets:14460 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:1265343 (1.2 MiB)  TX bytes:15901844 (15.1 MiB)

eth0      Link encap:Ethernet  HWaddr D4:6E:0E:32:C1:C2  
          inet addr:192.168.1.78  Bcast:192.168.1.255  Mask:255.255.255.0
          inet6 addr: 2602:306:8378:13b0:d66e:eff:fe32:c1c2/64 Scope:Global
          inet6 addr: fe80::d66e:eff:fe32:c1c2/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:17345 errors:0 dropped:0 overruns:0 frame:0
          TX packets:9783 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:16048234 (15.3 MiB)  TX bytes:1383170 (1.3 MiB)
          Interrupt:5 

eth1      Link encap:Ethernet  HWaddr D4:6E:0E:32:C1:C2  
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:9943 errors:0 dropped:0 overruns:0 frame:0
          TX packets:14465 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:1404545 (1.3 MiB)  TX bytes:15902270 (15.1 MiB)
          Interrupt:4 

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:26 errors:0 dropped:0 overruns:0 frame:0
          TX packets:26 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:2301 (2.2 KiB)  TX bytes:2301 (2.2 KiB)

apcameron commented:
@ae6xe were you able to verify that WIFI is functioning as expected?

apcameron commented:
@adrianschmutzler I have tested with your latest updates
TP-Link CPE610 v1 works as before 
TP-Link CPE510 v2 works as before 
TP-Link CPE510 v3 I do not have one to test currently
TP-Link CPE210 v1.1 works as expected
TP-Link CPE220 v2 Image sent to Eddie to Test.

ae6xe commented:
> @ae6xe were you able to verify that WIFI is functioning as expected?

Yes, wifi is functioning on cpe510 v1.1 as expected.
root@OpenWrt:~# iw dev wlan0 info
Interface wlan0
	ifindex 6
	wdev 0x2
	addr d4:6e:0e:32:c1:c2
	ssid OpenWrt
	type AP
	wiphy 0
	channel 36 (5180 MHz), width: 20 MHz, center1: 5180 MHz
	txpower 23.00 dBm
	multicast TXQ:
		qsz-byt	qsz-pkt	flows	drops	marks	overlmt	hashcol	tx-bytes	tx-packets
		0	0	151	0	0	0	0	27095		151
root@OpenWrt:~# iw phy phy0 info
Wiphy phy0
	max # scan SSIDs: 4
	max scan IEs length: 2261 bytes
	max # sched scan SSIDs: 0
	max # match sets: 0
	max # scan plans: 1
	max scan plan interval: -1
	max scan plan iterations: 0
	Retry short limit: 7
	Retry long limit: 4
	Coverage class: 0 (up to 0m)
	Device supports AP-side u-APSD.
	Device supports T-DLS.
	Available Antennas: TX 0x3 RX 0x3
	Configured Antennas: TX 0x3 RX 0x3
	Supported interface modes:
		 * IBSS
		 * managed
		 * AP
		 * AP/VLAN
		 * monitor
		 * mesh point
		 * P2P-client
		 * P2P-GO
		 * outside context of a BSS
	Band 2:
		Capabilities: 0x11ef
			RX LDPC
			HT20/HT40
			SM Power Save disabled
			RX HT20 SGI
			RX HT40 SGI
			TX STBC
			RX STBC 1-stream
			Max AMSDU length: 3839 bytes
			DSSS/CCK HT40
		Maximum RX AMPDU length 65535 bytes (exponent: 0x003)
		Minimum RX AMPDU time spacing: 8 usec (0x06)
		HT TX/RX MCS rate indexes supported: 0-15
		Frequencies:
			* 5180 MHz [36] (23.0 dBm)
			* 5200 MHz [40] (23.0 dBm)
			* 5220 MHz [44] (23.0 dBm)
			* 5240 MHz [48] (23.0 dBm)
			* 5260 MHz [52] (23.0 dBm) (radar detection)
			* 5280 MHz [56] (23.0 dBm) (radar detection)
			* 5300 MHz [60] (23.0 dBm) (radar detection)
			* 5320 MHz [64] (23.0 dBm) (radar detection)
			* 5500 MHz [100] (23.0 dBm) (radar detection)
			* 5520 MHz [104] (23.0 dBm) (radar detection)
			* 5540 MHz [108] (23.0 dBm) (radar detection)
			* 5560 MHz [112] (23.0 dBm) (radar detection)
			* 5580 MHz [116] (23.0 dBm) (radar detection)
			* 5600 MHz [120] (23.0 dBm) (radar detection)
			* 5620 MHz [124] (23.0 dBm) (radar detection)
			* 5640 MHz [128] (23.0 dBm) (radar detection)
			* 5660 MHz [132] (23.0 dBm) (radar detection)
			* 5680 MHz [136] (23.0 dBm) (radar detection)
			* 5700 MHz [140] (23.0 dBm) (radar detection)
			* 5745 MHz [149] (26.0 dBm)
			* 5765 MHz [153] (26.0 dBm)
			* 5785 MHz [157] (26.0 dBm)
			* 5805 MHz [161] (26.0 dBm)
			* 5825 MHz [165] (26.0 dBm)
	valid interface combinations:
		 * #{ managed } <= 2048, #{ AP, mesh point } <= 8, #{ P2P-client, P2P-GO } <= 1, #{ IBSS } <= 1,
		   total <= 2048, #channels <= 1, STA/AP BI must match, radar detect widths: { 20 MHz (no HT), 20 MHz, 40 MHz }

	HT Capability overrides:
		 * MCS: ff ff ff ff ff ff ff ff ff ff
		 * maximum A-MSDU length
		 * supported channel width
		 * short GI for 40 MHz
		 * max A-MPDU length exponent
		 * min MPDU start spacing
	Supported extended features:
		* [ RRM ]: RRM
		* [ CQM_RSSI_LIST ]: multiple CQM_RSSI_THOLD records
		* [ CONTROL_PORT_OVER_NL80211 ]: control port over nl80211
		* [ TXQS ]: FQ-CoDel-enabled intermediate TXQs

adrianschmutzler commented:
Based on feedback via mail, CPE220 V2 is confirmed to be working, too.

adschm committed:
ath79: add label-mac-device for several TP-Link CPExxx devices

All those devices only have one MAC address, which is also printed
on label (obviously).

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adrianschmutzler commented:
Update:
- Added the capitalization fixes to have "TP-Link" everywhere consistently
- Added label-mac-device in a separate patch

**This should be ready for merge now.**
(If label-mac-device is not desired at the current point, the patch can just be omitted and I will include it in #2159)

Tested on the following devices:
TP-Link CPE610 v1 @apcameron 
TP-Link CPE510 v2 @apcameron 
TP-Link CPE210 v1.1 @adrianschmutzler @apcameron 
TP-Link CPE220 v2 Eddie
TP-Link CPE510 v1 @ae6xe

adrianschmutzler commented:
Fixed commit title capitalization and did (trivial) rebase.

adrianschmutzler commented:
@blocktrron @ynezz @981213 
One of you interested in reviewing/merging this one? Should be in good shape ...

blocktrron commented on tools/firmware-utils/src/tplink-safeloader.c:
```diff
@@ -244,17 +284,7 @@ static struct device_info boards[] = {
-			"CPE510(TP-LINK|US|N300-5|55530000):2.0\r\n"
-			"CPE510(TP-LINK|UN|N300-5):2.0\r\n"
-			"CPE510(TP-LINK|EU|N300-5):2.0\r\n"
-			"CPE510(TP-LINK|US|N300-5):2.0\r\n",
```
Say high to ar71xx:

https://github.com/openwrt/openwrt/commit/bae927c551fdd91fbc8343071e4b5c44f73eee0d#diff-b232b41a358a2b44c64fb32925bd265a

The removal of the CPE510 v2 from the v1 Supportlist will brick devices in ar71xx when upgrading from an earlier build.

Honestly, as ar71xx is source only now, i would (personally) be finde dropping the CPE510 v2 from here. But as it doesn't seem to hurt us much, maybe exclude the removal and place a short comment explaining the situation. 

What do you think?

adrianschmutzler replied:
Ah, I didn't find the commit you referenced in the first place.

Several things cross my mind here:
1. Removing v2 from CPE510 v1 support list already is a separate patch in this PR, so there won't be interference with the rest no matter what we decide.
2. I also change ar71xx when adjusting kernel partitions. I don't think this should be backported, so this will already introduce differences between 19.07 and master anyway. In this context, it does not seem too wrong to touch CPE510v2 support , too.
3. Having the link to the initial patch introducing v2 "support", I'm currently in favor of just reverting that patch entirely. It was a hack in the first place, and reverting the patch will remove support (of v2 in master/ar71xx) more cleanly than just changing the safeloader data.
4. I'm aware that this complicates backporting if we ever have to fix something with CPE510. Nevertheless, not removing lines in safeloader now will "enable" flashing CPE510v2@ath79 with v1 images, won't it? So, I would argue to keep the master clean (and remove CPE510v2@ar71xx). Fixes for ar71xx@19.07 can still be done by just basing patches directly on it.
5. What do you mean by "maybe exclude the removal"?

So, I would:
1. Apply this patchset WITHOUT the patch removing v2 from safeloader.
2. Remove the v2 support afterwards with "git revert bae927c". It might be easier if you just do it, if you do not want or have time I can also provide the patch.
3. Change the SUPPORTED_DEVICES for CPE510v2 in ath79 to "cpe510" as the current "cpe510-v2" seems to be wrong then. Again, you can just do that during merging this PR or I can send a small patch for this one afterwards.

adrianschmutzler replied:
(If you do prefer that, I can obviously also just add a bigger explanation to the existing patch)

blocktrron replied:
Personally, i would leave the SupportList for the CPE510v1 untouched and simply place a note above it, that the CPE510v2 entries can be deleted after ar71xx is deprecated.

> I also change ar71xx when adjusting kernel partitions. I don't think this should be backported, so this will already introduce differences between 19.07 and master anyway. In this context, it does not seem too wrong to touch CPE510v2 support , too.

The kernel partition change is no change which possibly breaks updates here. ar71xx is source only, but I'm uncertain if we should really break compatibility now (or invest a lot of time in working around this issue).

> Having the link to the initial patch introducing v2 "support", I'm currently in favor of just reverting that patch entirely. It was a hack in the first place, and reverting the patch will remove support (of v2 in master/ar71xx) more cleanly than just changing the safeloader data.

I agree, the patch is a hacky way of adding support. However, ar71xx is full of those hacks (See the TL-WR841 mess). The Question of dropping support for the v2 is something I wouldn't want to decide alone. 

> Nevertheless, not removing lines in safeloader now will "enable" flashing CPE510v2@ath79 with v1 images, won't it?

Yes, however (given we provide both separate images), this shouldn't be an issue. Flashing the "wrong image" is never a good idea, so it is in this case.

> So, I would argue to keep the master clean (and remove CPE510v2@ar71xx). Fixes for ar71xx@19.07 can still be done by just basing patches directly on it.

Again, this comes close to a removal of the CPE510v2 in ar71xx. This is something you should probably get opinions on via the ML. Yes, i think I'm "to new" to decide something like this alone :confused: I hope you understand this.

> What do you mean by "maybe exclude the removal"?

To remove https://github.com/openwrt/openwrt/pull/2252/commits/1d7326439d797da8966f5257b12053bb760cb1b9 from this PR.

adrianschmutzler replied:
Okay, just drop the patch then.
I will think about whether the CPE510v2 topic is important enough to move it to the mailing list.

apcameron commented:
PR #1631 was created to give the CPE510 V2 its own profile but closed when I was told to switch to the ATH79 Branch.

adrianschmutzler commented:
@apcameron I will have a look at #1631 . Maybe one can boil it down to something that's small enough to be accepted as a fix for the current hack in ar71xx. Maybe it's too much work and we just leave it as it is.

apcameron commented:
I Agree the best approach is to remove 1d73264 from this PR.

adrianschmutzler commented:
I've removed the patch and rebased on latest master.

adrianschmutzler commented:
I added a patch for the CPE510v2 issue in #2302 
In my personal opinion changes to ar71xx introduced there are small enough to do this even at the current situation (with the benefit of getting tplink-safeloader.c clean).

blocktrron commented:
Checkpatch found the following misspells:

```
--------------------------------------------------
0004-ath79-add-support-for-TP-Link-CPE210-v1.patch
--------------------------------------------------
WARNING: 'adress' may be misspelled - perhaps 'address'?
#22: 
Stock device TFTP adress:192.168.0.254

--------------------------------------------------
0005-ath79-add-support-for-TP-Link-CPE510-v1.patch
--------------------------------------------------
WARNING: 'adress' may be misspelled - perhaps 'address'?
#25: 
Stock device TFTP adress:192.168.0.254

0006-ath79-add-support-for-TP-Link-CPE220-v2.patch
--------------------------------------------------
WARNING: 'adress' may be misspelled - perhaps 'address'?
#25: 
Stock device TFTP adress:192.168.0.254

```

Otherwise, this looks good to me.

adrianschmutzler commented:
fixed "address" typo (3 times).

blocktrron commented:
@adrianschmutzler I've pulled your patches except for https://github.com/openwrt/openwrt/pull/2252/commits/ffd49dd0635db75cbaab13a6e93e21428b7389bf into my staging tree, as this concept has still the general RFC PR open.

I propose you add this patch there after I've pushed your other patches to master. Is this fine for you?

adrianschmutzler commented:
@blocktrron Yes, I will just add the label-mac-device to my label MAC PR. It was just included here so I do not forget...

blocktrron commented:
Okay, closing then.
